pbsmetrics/metrics.go

```diff
@@ -113,7 +115,7 @@ const (
 // two groups should be consistent within themselves, but comparing numbers between groups
 // is generally not useful.
 type MetricsEngine interface {
-	RecordRequest(labels Labels)                           // ignores adapter. only statusOk and statusErr fom status
+	RecordRequest(labels Labels, numImps int)              // ignores adapter. only statusOk and statusErr fom status
```

Comment:
Makes me second guess if the `RecordRequest` and `RecordAdapterRequest` sets of methods should be condensed into just the two functions or not. The split is based on each function storing one piece of event data, which in a tag based metrics system would be one metric with tags attached. Now `RecordRequest` is recording 2 metrics, requests, and the number of imps delivered. I think splitting them is simpler conceptually, but your example suggests condensing the types together may improve extensibility.

Not going to say anything is wrong with this PR, but gives food for thought for future enhancements and cleanup.

Reply:
hmmm... I actually agree with this.

I don't know what the right path forward is... but I do know I should follow your design until we decide to change it. It's obvious in retrospect, but I didn't realize I was breaking convention.

Is fixed now.